### PR TITLE
Add an IsTaskDone util to the history queues

### PR DESCRIPTION
**Pull request by MichaelSnowden**
This is a utility which will generalize the queue-checks that are used in https://github.com/temporalio/temporal/issues/3368

### CODE REVIEW

File: service/history/queues/util.go [hunk 1]
```diff
@@ -56,3 +58,20 @@ func IsTaskAcked(
...
+		return true
+	}
+	persistenceQueueState, ok := shard.GetQueueState(category)
+	if !ok {
+		// Use global ack level visibility queue ack level because cluster level is not updated.
+		visibilityQueueAckLevel := shard.GetQueueAckLevel(category).TaskID
+		return taskID <= visibilityQueueAckLevel
+	}
+
+	fakeTask := tasks.NewFakeTask(workflowKey, category, tasks.DefaultFireTime)
```
Comment (yycptt) [inline service/history/queues/util.go:74]: We can't use fake task here, task type information is necessary for `IsTaskAcked` to determine if a task is still pending or not.

Comment (yycptt) [reply service/history/queues/util.go:74]: Task fire time information is also required, so defaultFireTime probably can't work.

Comment (MichaelSnowden) [reply service/history/queues/util.go]: Fixed

**Comment by MichaelSnowden** [comment]
After discussing this with @yycptt , seems like there isn't much of a point in generalizing this

**Pull request closed without merging**